Realm/Realm/Extensions/CollectionExtensions.cs

```diff
@@ -365,6 +403,42 @@ public static IQueryable<T> Filter<T>(this IList<T> list, string predicate, para
             return realmList.GetFilteredResults(predicate, arguments);
         }
 
+        /// <summary>
+        /// Apply an NSPredicate-based filter over a collection. It can be used to create
+        /// more complex queries, that are currently unsupported by the LINQ provider and
+        /// supports SORT and DISTINCT clauses in addition to filtering.
+        /// </summary>
+        /// <typeparam name="T">The type of the objects that will be filtered.</typeparam>
+        /// <param name="set">A Realm Set.</param>
+        /// <param name="predicate">The predicate that will be applied.</param>
+        /// <param name="arguments">Values used for substitution in the predicate. Note that all primitive types are accepted as they are implicitly converted to RealmValue.</param>
+        /// <returns>A queryable observable collection of objects that match the predicate.</returns>
+        /// <remarks>
+        /// This method can be used in combination with LINQ filtering, but it is strongly recommended
+        /// to avoid combining it if a <c>SORT</c> clause appears in the predicate.
+        /// <para/>
+        /// If you're not going to apply additional filters, it's recommended to use <see cref="AsRealmCollection{T}(IQueryable{T})"/>
+        /// after applying the predicate.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// var results1 = realm.All&lt;Foo&gt;("Bar.IntValue > 0");
+        /// var results2 = realm.All&lt;Foo&gt;("Bar.IntValue > 0 SORT(Bar.IntValue ASC Bar.StringValue DESC)");
+        /// var results3 = realm.All&lt;Foo&gt;("Bar.IntValue > 0 SORT(Bar.IntValue ASC Bar.StringValue DESC) DISTINCT(Bar.IntValue)");
```

Comment:
These code snippets are wrong - I had copy-pasted them in the IList API docs and forgotten to update them. Can you go through both and update them to something meaningful?